bigquery/src/main/scala/com/yahoo/maha/executor/bigquery/BigqueryQueryExecutor.scala

```diff
@@ -0,0 +1,174 @@
+// Copyright 2017, Yahoo Holdings Inc.
+// Licensed under the terms of the Apache License 2.0. Please see LICENSE file in project root for terms.
+package com.yahoo.maha.executor.bigquery
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.scala.{DefaultScalaModule, ScalaObjectMapper}
+import com.github.vertical_blank.sqlformatter.scala.{FormatConfig, SqlDialect, SqlFormatter}
+import com.google.auth.oauth2.ServiceAccountCredentials
+import com.google.cloud.ServiceOptions
+import com.google.cloud.bigquery.BigQuery
+import com.google.cloud.bigquery.BigQueryOptions
+import com.google.cloud.bigquery.JobId
+import com.google.cloud.bigquery.JobInfo
+import com.google.cloud.bigquery.QueryJobConfiguration
+import com.yahoo.maha.core._
+import com.yahoo.maha.core.query._
+import grizzled.slf4j.Logging
+import java.io.FileInputStream
+import java.net.{Authenticator, PasswordAuthentication}
+import java.util.UUID
+import scala.util.Try
+
+private case class ProxyCredentials(
+  @JsonProperty("proxy_user") userName: String,
+  @JsonProperty("proxy_pass") password: String
+)
+
+case class BigqueryQueryExecutorConfig(
+  gcpCredentialsFilePath: String,
+  gcpProjectId: String,
+  enableProxy: Boolean,
+  proxyCredentialsFilePath: Option[String],
+  proxyHost: Option[String],
+  proxyPort: Option[String],
+  retries: Int
+) extends Logging {
+
+  private final val HttpsProxyHost = "https.proxyHost"
+  private final val HttpsProxyPort = "https.proxyPort"
+  private final val DisabledAuthSchemes = "jdk.http.auth.tunneling.disabledSchemes"
+
+  private val mapper = {
+    val om = new ObjectMapper(new YAMLFactory()) with ScalaObjectMapper
+    om.registerModule(DefaultScalaModule)
+    om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    om
+  }
+
+  def buildBigqueryClient(): BigQuery = {
+    BigQueryOptions
+      .newBuilder()
+      .setCredentials(ServiceAccountCredentials.fromStream(new FileInputStream(gcpCredentialsFilePath)))
+      .setProjectId(gcpProjectId)
+      .setRetrySettings(ServiceOptions.getDefaultRetrySettings().toBuilder().setMaxAttempts(retries).build())
+      .build()
+      .getService
+  }
+
+  def configureProxy(): Unit = {
+    if (enableProxy && proxyCredentialsFilePath.isDefined && proxyHost.isDefined && proxyHost.isDefined) {
+      val proxyCredentials = mapper.readValue[ProxyCredentials](
+        new FileInputStream(proxyCredentialsFilePath.get)
+      )
+
+      System.setProperty(HttpsProxyHost, proxyHost.get)
+      System.setProperty(HttpsProxyPort, proxyPort.get)
+      System.setProperty(DisabledAuthSchemes, "")
+
+      Authenticator.setDefault(new Authenticator() {
```

Comment:
Hm, is this a global?  Is there a way to isolate it for use only by this component?

Reply:
Switched from the global Authenticator to setting the proxy auth as transport options in the Google API client.